integration-tests/source-test.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/projectdiscovery/folderutil"
+	"github.com/projectdiscovery/uncover/testutils"
+)
+
+var (
+	ConfigFile = filepath.Join(folderutil.HomeDirOrDefault("."), ".config/uncover/provider-config.yaml")
+)
+
+type censysTestcases struct{}
+
+func (h censysTestcases) Execute() error {
+	token := os.Getenv("CENSYS_API_KEY")
+	if token == "" {
+		return errors.New("missing censys api key")
+	}
+	censysToken := fmt.Sprintf(`censys: [%s]`, token)
+	_ =ioutil.WriteFile(ConfigFile, []byte(censysToken), 0644)
```

Comment:
`ioutil.WriteFile` => `os.WriteFile` (`ioutil` is deprecated)

Reply:
Sure.